WcaOnRails/app/assets/javascripts/currencies.js

```diff
@@ -0,0 +1,77 @@
+window.wca = window.wca || {};
+
+wca.getCurrencyInfo = function(isoCode) {
+  return wca._currenciesInfo[isoCode] || wca._currenciesInfo.USD;
+};
+
+// Create a mask for an amount input in the given currency, using autoNumeric
+// 'action' can be either "init" or "update"
+// 'elemId' is a selector for the targeted input field
+wca.applyCurrencyMask = function(action, elemId, currencyIsoCode) {
+  if (!$(elemId).length)
+    return;
+  var entry = wca.getCurrencyInfo(currencyIsoCode);
+  var currentVal = 0;
+
+  // Get current val
+  if (action == "update") {
+    currentVal = wca.getValueInCurrency(elemId);
+  } else if (action == "init") {
+    currentVal = $(elemId).val();
+  } else {
+    return;
```

Comment:
This also scares me. If we pass in an invalid action, how are we going to notice? Should we throw an expception here instead?